### ISSUE TITLE
Fix `property-no-unknown` false positives for descriptors

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#8155]
### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "property-no-unknown": true
  }
}
```

### Do you have a proposal to fix the bug?

1. **#8148 should be completed**
2. exceptions like
https://github.com/stylelint/stylelint/blob/4385d8fcbf47936f6a16236548fe106b2f9a99a3/lib/rules/property-no-unknown/index.mjs#L26-L27 
should be removed
i.e. decline #7751
3. an at-rule/descriptor map should be created for the descriptors that happen to be named after existing properties
4. the descriptors should be ignored
